packages/mcp/src/index.ts

```diff
@@ -0,0 +1,174 @@
+      throw new Error(
+        `Invalid tool: ${tool}. Accepted tools are: ${ACCEPTED_TOOLS.join(
+          ", ",
+        )}`,
+      );
+    }
+  });
+
+  // Check if API key is either provided in args or set in environment variables
+  const privateKey = options.privateKey || RECALL_PRIVATE_KEY;
+  if (!privateKey) {
+    throw new Error(
+      "Recall private key not provided. Please either pass it as an argument --private-key=$KEY or set the RECALL_PRIVATE_KEY environment variable.",
+    );
+  }
+  options.privateKey = privateKey;
+
+  // If not set, the `RecallAgentToolkit` will default to `testnet`
+  options.network = options.network || RECALL_NETWORK;
+  if (
+    options.network &&
+    options.network !== "testnet" &&
+    options.network !== "localnet"
+  ) {
+    throw new Error(
+      "Invalid network. Please use either 'testnet' or 'localnet'.",
+    );
+  }
+
+  return options;
+}
+
+// Handle errors with special color formatting for clarity purposes since we use console.error
+// for all logging outputs
+function handleError(error: unknown) {
+  console.error(chalk.red("\nError initializing Recall MCP server:\n"));
+  console.error(
+    chalk.yellow(
+      ` ${error instanceof Error ? error.message : String(error)}\n`,
+    ),
+  );
+}
+
+// Main function that parses the command line arguments, creates the `RecallAgentToolkit` instance,
+// and starts the MCP server
+export async function main() {
+  const options = parseArgs(process.argv.slice(2));
+
+  // Create the `RecallAgentToolkit` instance
+  const selectedTools = options.tools!;
+  const configuration: Configuration = { actions: {}, context: {} };
+
+  if (selectedTools.includes("all")) {
+    ACCEPTED_TOOLS.forEach((tool) => {
+      const [resource, action] = tool.split(".");
+      configuration.actions[resource as Resource] = {
+        ...configuration.actions[resource as Resource],
+        [action as Permission]: true,
+      };
+    });
+  } else {
+    selectedTools.forEach((tool) => {
+      const [resource, action] = tool.split(".");
+      if (!resource || !action) {
+        throw new Error(
+          `Invalid tool: ${tool}. Tool must be in the format of "resource.action".`,
+        );
+      }
+      configuration.actions[resource as Resource] = {
+        ...configuration.actions[resource as Resource],
+        [action]: true,
+      };
+    });
+  }
+
+  // Append custom Recall network to configuration if provided
+  if (options.network) {
+    configuration.context = { network: options.network };
+  }
+
+  const server = new RecallAgentToolkit({
+    privateKey: options.privateKey!,
+    configuration: configuration,
+  });
+
+  const transport = new StdioServerTransport();
+  await server.connect(transport);
+  // We use console.error instead of console.log since console.log will output to stdio, which
+  // will confuse the MCP server
+  console.error(chalk.green("Recall MCP Server running on stdio"));
+}
+
+main().catch((error) => {
```

Comment:
If possible, I'd rather use a top-level await within try/catch. But that might not work depending on the node version?

Reply:
hm so our root package file defines node >=20 as the supported version. we [inherit the `module` definition](https://github.com/recallnet/js-recall/blob/b5833f9c24d8ca184c4d318bede8708247aba402/packages/typescript-config/base.json#L11) as NodeNext. so, it's probably fine?

Reply:
naw, don't change anything. anyone writing agents is probably using pretty new versions of stuff